src/metabase/api/dataset.clj

```diff
@@ -172,4 +175,56 @@
     (qp.streaming/streaming-response [context :api]
       (qp.pivot/run-pivot-query (assoc query :async? true) info context))))
 
+
+(def ^:private Parameter
+  (su/open-schema
+   {:values_source_type (s/enum nil "static-list" "card")}))
+
+(def ^:private FieldIds (s/maybe [s/Int]))
+
+(defn- parameter-field-values
+  [field-ids query]
+  (reduce (fn [resp id]
+            (let [{values :values more? :has_more_values} (api.field/field-id->values id query)]
+              (-> resp
+                  (update :values concat values)
+                  (update :has_more_values #(or % more?)))))
+          {:has_more_values false
+           :values []}
+          field-ids))
+
+(defn parameter-values
+  "Fetch parameter values. Parameter should be a full parameter, field-ids is an optional vector of field ids, only
+  consulted if `:values_source_type` is nil. Query is an optional string return matching field values not all."
+  [parameter field-ids query]
+  (case (:values_source_type parameter)
+    "static-list" (params.static-values/param->values parameter query)
+    "card"        (params.card-values/param->values parameter query)
+    nil           (if (seq field-ids)
+                    (parameter-field-values field-ids query)
+                    (throw (ex-info (tru "Missing field-ids for parameter")
+                                    {:status-code 400
+                                     :parameter parameter})))
+    (throw (ex-info (tru "Invalid parameter source {0}" (:values_source_type parameter))
+                    {:status-code 400
+                     :parameter parameter}))))
+
+#_{:clj-kondo/ignore [:deprecated-var]}
+(api/defendpoint-schema POST "/parameter/values"
+  "Return parameter values for cards or dashboards that are being edited."
+  [:as {{:keys [parameter field_ids]} :body}]
+  {parameter Parameter
+   field_ids  FieldIds}
+  (let [nil-query nil]
+    (parameter-values parameter field_ids nil-query)))
+
+#_{:clj-kondo/ignore [:deprecated-var]}
+(api/defendpoint-schema POST "/parameter/search"
```

Comment:
@dpsutton Could we put the query into the url as everywhere else? e.g. https://github.com/metabase/metabase/blob/d369e9408d0cd45aa372d23e8c67611e893bbaf7/src/metabase/api/public.clj#L484

Reply:
good catch. Fixed.